### PR TITLE
Fix: jazzy docs for swift

### LINKED ISSUE #707: Configure Jazzy to generate docs for public Swift classes
`jazzy` is currently set up to only generate docs for our `Purchases` objective-c module. 
We should configure it to also generate docs for swift modules. 
Since a lot of our swift stuff has been made public only temporarily to aid in the transition to objective-c, the configuration should be set up so that it only reads from the `Public` folder in `PurchasesCoreSwift`

### PR DESCRIPTION
**Pull request by aboedo**
Jazzy docs weren't running on the Swift target, so we were missing out on documenting all the new stuff in Swift. 

Jazzy doesn't strictly support documenting multiple modules, or even supporting mixed-language targets, but they do have some documentation on [how to do it anyway](https://github.com/realm/jazzy#mixed-objective-c--swift), and it works correctly. And we're not the only ones doing this, [since the Facebook iOS SDK does this as well.](https://github.com/facebook/facebook-ios-sdk/blob/69f0766c149282277528dc59917454cc3d1cc880/scripts/genDocs.rb)

This does mean adding a new script to call jazzy manually, since this requires going down to SourceKitten and a couple of commands. 

Tested locally and the docs are generated correctly with public stuff **only from the public folders** from both modules. 

Fixes #707

### CODE REVIEW

File: scripts/generate_docs.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    -I $(pwd) -fmodules > objcDoc.json
+
+# Feed both outputs to Jazzy as a comma-separated list
+jazzy --sourcekitten-sourcefile swiftDoc.json,objcDoc.json \
+    --module Purchases \
+    --output generated_docs/
+
+# clean up generated files
+rm swiftDoc.json
+rm objcDoc.json
```
Comment (aboedo) [inline scripts/generate_docs.sh:19]: not strictly needed, but I don't like having extra files lying around

File: .gitignore [hunk 2]
```diff
@@ -32,4 +32,5 @@ CarthageUploads
 builds/
 Pods/
 IntegrationTests/CarthageIntegration/Cartfile.resolved
-scan_derived_data/
\ No newline at end of file
+scan_derived_data/
+generated_docs/
```
Comment (aboedo) [inline .gitignore:36]: we didn't use to have the docs folder generated, but it gets annoying when you run jazzy locally and end up with an uncommitted but not ignored folder lying around

File: .jazzy.yaml [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
-objc: true
 sdk: iphonesimulator
-module: Purchases
-umbrella_header: Purchases/Public/Purchases.h
 module_version: 3.13.0-SNAPSHOT
 github_url: https://github.com/revenuecat/purchases-ios
 github_file_prefix: https://github.com/revenuecat/purchases-ios/tree/3.13.0-SNAPSHOT
 output: docs
-# Leaving this commented out. We used to specify this before, but now it's working without it
-# xcodebuild_arguments: [--objc,Purchases/Public/Purchases.h,--,-x,objective-c,-isysroot,$(xcrun --show-sdk-path),-I,$(pwd)]
```
Comment (aboedo) [inline .jazzy.yaml:13]: these docs haven't been relevant for a while, and they weren't particularly informative

File: .jazzy.yaml [hunk 4]
```diff
@@ -1,13 +1,11 @@
 clean: true
 author: RevenueCat
 author_url: https://revenuecat.com
-objc: true
 sdk: iphonesimulator
-module: Purchases
-umbrella_header: Purchases/Public/Purchases.h
```
Comment (aboedo) [inline .jazzy.yaml:7]: no longer objective-c-only, so I'm removing these unnecessary params

File: fastlane/Fastfile [hunk 5]
```diff
@@ -245,6 +245,13 @@ platform :ios do
     }
   end
 
+  desc "Generate Jazzy Docs"
+  lane :generate_docs do
+    Dir.chdir("..") do
+      sh("./scripts/generate_docs.sh")
```
Comment (aboedo) [inline fastlane/Fastfile:251]: we could just run this directly, but I figured having a lane would be a nice convenience

**Review by taquitos — APPROVED** [review]
🎈🐐

**Pull request merged**